manifest.json

```diff
@@ -36,6 +38,13 @@
         "pollInterval": {
           "description": "The interval, in seconds, at which to poll property values.",
           "type": "number"
+        },
```

Comment:
I'd prefer this to not be a separate array. Maybe a better option is to have optional fields in the `urls` array for username/password, JWT, etc.

Reply:
Did you mean this bit? 
`"secureThings": {
          "description": "Hostname and JWT to be used to access secureThings. Separate by a space.",
          "type":"array",
          "items": {
            "type": "string"
          }
        }`

Can you give me an example of what this might look like? I could certainly find them in the array

Perhaps something like 
["jwt raspib.local:8888 the_jwt", "url the_url"] 
it would be easy enough to sort that. 
 
Also -- could you point me to where the format of the manifest.json is documented? I couldn't find anything when I searched so I just copied the url entry.

Reply:
I'm thinking something like this:
```json
  "options": {
    "default": {
      "urls": [],
      "pollInterval": 5
    },
    "schema": {
      "type": "object",
      "required": [
        "urls",
        "pollInterval"
      ],
      "properties": {
        "urls": {
          "type": "array",
          "items": {
            "type": "object",
            "required": [
              "href"
            ],
            "properties": {
              "href": {
                "description": "Base URL of web thing",
                "type": "string"
              },
              "authentication": {
                "description": "Authentication credentials",
                "type": "object",
                "required": [
                  "method"
                ],
                "properties": {
                  "method": {
                    "type": "string",
                    "enum": [
                      "basic",
                      "digest",
                      "jwt"
                    ]
                  },
                  "username": {
                    "description": "Username (for basic and digest)",
                    "type": "string"
                  },
                  "password": {
                    "description": "Password (for basic and digest)",
                    "type": "string"
                  },
                  "realm": {
                    "description": "Realm (for digest)",
                    "type": "string"
                  },
                  "token": {
                    "description": "Token (for JWT)",
                    "type": "string"
                  }
                }
              }
            }
          }
        },
        "pollInterval": {
          "description": "The interval, in seconds, at which to poll property values.",
          "type": "number"
        }
      }
    }
  },
```

The manifest format is documented [here](https://github.com/WebThingsIO/addon-list/blob/master/manifest.md).

Reply:
I commited a pass at this implementation. It works with my jwt stuff. Do you have a way to test basic and digest? 
I did not implement basic and digest, but I put stubs in to add later. 
I'd have to get a thing behind nginx to test the other stuff with.

Check it out:

![Screen Shot 2020-09-29 at 7 34 26 PM](https://user-images.githubusercontent.com/2983414/94636787-ce9bee00-028a-11eb-82aa-ea59a13c6604.png)

Reply:
Feel free to just stub the basic and digest parts in. I can finish that part up.